internal/hcl/parser.go

```diff
@@ -397,9 +397,28 @@ func (p *Parser) ParseDirectory() (m *Module, err error) {
 		nil,
 	)
 
-	root, err := evaluator.Run()
-	if err != nil {
-		return m, err
+	var root *Module
+
+	// Graph evaluation
+	if os.Getenv("INFRACOST_GRAPH_EVALUATOR") == "true" {
+		// we use the base logrus log here so that it's consistent with the spinner logs
+		logrus.Info("Building project with experimental graph runner")
+
+		g, err := NewGraphWithRoot(p.logger, nil)
+		if err != nil {
+			return m, err
+		}
+
+		root, err = g.Run(evaluator)
+		if err != nil {
+			return m, err
+		}
+	} else {
+		// Existing evaluation
+		root, err = evaluator.Run()
+		if err != nil {
+			return m, err
+		}
 	}
```

Comment:
@hugorut should we move all this logic inside evaluator.Run?

This would also fix the (not-important) issue that we're not seeing the ⠴ Evaluating Terraform directory spinner when using the graph evaluator.

Reply:
I seems a bit wierd to have the `Graph` -> which has a dep on evaluator, to be initialised and used within the `Graph`. I think it makes actually more sense to place it here, with the understanding that `Evaluator.Run` will be removed in the future when we migrate fully to the `Graph`. If you think it cleans up some stuff though, I'm not super against it so feel free to move, but add a comment explaining the whys/whats.